### ISSUE TITLE
Unrecognized selector sent to instance

### ISSUE THREAD

**Issue by GorkaMM** [Notificare/notificare-push-lib#3]
Since Notificare 1.4.0, our iOS app is crashing to some users. This is the full trace:

**Fatal Exception: NSInvalidArgumentException**
`-[UIConcreteLocalNotification objectForKey:]: unrecognized selector sent to instance 0x15e66940`

```
Thread : Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x26ef749f __exceptionPreprocess + 126
1  libobjc.A.dylib                0x346adc8b objc_exception_throw + 38
2  CoreFoundation                 0x26efc8b9 __methodDescriptionForSelector
3  CoreFoundation                 0x26efa7d7 ___forwarding___ + 714
4  CoreFoundation                 0x26e2c058 _CF_forwarding_prep_0 + 24
5  (appname)                      0x0019aeef -[NotificarePushLib handleOptions:] (NotificarePushLib.m:433)
6  (appname)                      0x000ce177 -[UPCAppDelegate application:didFinishLaunchingWithOptions:] (UPCAppDelegate.m:53)
7  UIKit                          0x2a4244f1 -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 284
8  UIKit                          0x2a61943f -[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 2358
9  UIKit                          0x2a61b98b -[UIApplication _runWithMainScene:transitionContext:completion:] + 1374
10 UIKit                          0x2a626209 __84-[UIApplication _handleApplicationActivationWithScene:transitionContext:completion:]_block_invoke + 36
11 UIKit                          0x2a61a217 -[UIApplication workspaceDidEndTransaction:] + 130
12 FrontBoardServices             0x2d6880d1 __31-[FBSSerialQueue performAsync:]_block_invoke + 12
13 CoreFoundation                 0x26ebdd7d __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
14 CoreFoundation                 0x26ebd041 __CFRunLoopDoBlocks + 216
15 CoreFoundation                 0x26ebb7c3 __CFRunLoopRun + 762
16 CoreFoundation                 0x26e093c1 CFRunLoopRunSpecific + 476
17 CoreFoundation                 0x26e091d3 CFRunLoopRunInMode + 106
18 UIKit                          0x2a41e1bf -[UIApplication _run] + 558
19 UIKit                          0x2a418fa1 UIApplicationMain + 1440
20 upclose                        0x000a448d main (main.m:16)
21 libdyld.dylib                  0x34c2daaf start + 2
```

**Comment by silentjohnny** [comment]
@GorkaMM 

It looks like the handleOptions chokes on a launch from a local notification. Can you confirm that your app schedules its own local notifications? This would limit the possible causes.

**Comment by GorkaMM** [comment]
No, my app doesn't schedule any local notifications. Hope this helps :)

Thank you, @silentjohnny!

**Comment by jjonthemix** [comment]
Hi Gorka, we've just released 1.4.1 that addresses this issue. This can only happen if launchWithOptions is invoked and UIApplicationLaunchOptionsLocalNotificationKey is not nil. If that would happen it would indeed generate this crash. 

We corrected this by ignoring any local notification included in the UIApplicationLaunchOptionsLocalNotificationKey that does not contain a valid Notificare notification Id.  If userInfo object of a local notification contains a property "id" with a valid notification Id we will handle it like we handle a remote notification. 

Hope this helps.
